sklearn/tree/_tree.pyx

```diff
@@ -217,8 +214,7 @@ cdef class DepthFirstTreeBuilder(TreeBuilder):
                     impurity = splitter.node_impurity()
                     first = 0
 
-                is_leaf = (is_leaf or
-                           (impurity <= min_impurity_split))
+                is_leaf = is_leaf or impurity == 0
```

Comment:
Doesn't this imply `min_impurity_split = 0` and also that the minimum impurity is 0?

Reply:
I don't see why it would be a problem: if a node is pure (0 entropy, GINI or MSE)  then there is no need to try to split it anymore and therefore we can flag it as a leaf.

Reply:
Shouldn't this be captured with the impurity improvement?
I'm asking because the poisson splitting criterion would be a bit faster if it is not required to have it's minimum at 0.

Reply:
I see, it is about pure nodes, i.e. all samples in the leaf have the same value. This then safes trying to split any further as opposed to impurity improvement.